src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java

```diff
@@ -30,81 +38,108 @@ public PostgresSQLConnection(DbConnection conn) {
 
   @Override
   public SQLConnection setAutoCommit(boolean autoCommit, Handler<AsyncResult<Void>> handler) {
-    throw new UnsupportedOperationException();
+    inAutoCommit.set(autoCommit);
```

Comment:
so the problem is more than when `autocommit` is set to false and we are in a transaction then the `SQLConnection` either not be usable (i.e throw IllegalStateException) or the evaluation of `beginTxIfNeeded` should be delayed.